spec/workflow/reference_path_spec.rb

```diff
@@ -1,7 +1,78 @@
 RSpec.describe Floe::Workflow::ReferencePath do
-  context "with invalid value" do
-    it "raises an exception" do
-      expect { described_class.new("$.foo@.bar") }.to raise_error(Floe::InvalidWorkflowError, "Invalid Reference Path")
+  let(:subject) { described_class.new(payload) }
+
+  describe "#initialize" do
+    context "with invalid value" do
+      let(:payload) { "$.foo@.bar" }
+
+      it "raises an exception" do
+        expect { subject }.to raise_error(Floe::InvalidWorkflowError, "Invalid Reference Path")
+      end
+    end
+  end
+
+  describe "#get" do
+    context "with a simple path" do
+      let(:payload) { "$" }
+      let(:input)   { {"hello" => "world"} }
+
+      it "returns the input" do
+        expect(subject.get(input)).to eq(input)
+      end
+    end
+
+    context "with an array dereference" do
+      let(:payload) { "$['store'][0]['book']" }
+      let(:input)   { {"store" => [{"book" => "ASL For Dummies"}]} }
```

Comment:
0 seems like an easy case.
For some reason, I want to see 1 working

```suggestion
      let(:payload) { "$['store'][1]['book']" }
      let(:input)   { {"store" => [{},{"book" => "ASL For Dummies"}]} }
```

Reply:
:+1: done (still works)